packages/dx-react-scheduler-material-ui/src/templates/view-switcher/switcher.jsx

```diff
@@ -59,13 +59,18 @@ const SwitcherBase = ({
 SwitcherBase.propTypes = {
   onChange: PropTypes.func.isRequired,
   classes: PropTypes.object.isRequired,
-  currentViewName: PropTypes.string,
-  availableViewNames: PropTypes.arrayOf(PropTypes.string),
+  currentView: PropTypes.shape({
+    name: PropTypes.string.isRequired,
+    displayName: PropTypes.string.isRequired,
+  }).isRequired,
+  availableViews: PropTypes.arrayOf(PropTypes.shape({
+    name: PropTypes.string.isRequired,
+    displayName: PropTypes.string.isRequired,
+  })),
 };
 
 SwitcherBase.defaultProps = {
-  currentViewName: undefined,
-  availableViewNames: [],
+  availableViews: [],
```

Comment:
If you specify a required property you should not define `defaultProps`

Reply:
It's not required property